test/unit/error.test.ts

```diff
@@ -1,53 +1,79 @@
+import {
   MongoError,
+  MongoNetworkError,
+  MongoParseError,
   MongoServerError,
-  MongoParseError
-} = require('../../src/index');
-const {
-  LEGACY_NOT_WRITABLE_PRIMARY_ERROR_MESSAGE,
-  LEGACY_NOT_PRIMARY_OR_SECONDARY_ERROR_MESSAGE,
-  NODE_IS_RECOVERING_ERROR_MESSAGE,
-  isRetryableEndTransactionError,
-  isSDAMUnrecoverableError
-} = require('../../src/error');
-const {
-  PoolClosedError: MongoPoolClosedError,
-  WaitQueueTimeoutError: MongoWaitQueueTimeoutError
-} = require('../../src/cmap/errors');
+  MongoWriteConcernError,
+  TopologyDescription
+} from '../../src/index';
+import * as importsFromEntryPoint from '../../src/index';
+import { Topology } from '../../src/sdam/topology';
+import { isHello, ns, setDifference } from '../../src/utils';
+import { ReplSetFixture } from '../tools/common';
+import { cleanup } from '../tools/mongodb-mock/index';
+import { getSymbolFrom } from '../tools/utils';
 
 describe('MongoErrors', () => {
-  // import errors as object
-  let errorClasses = Object.fromEntries(
-    Object.entries(require('../../src/index')).filter(([key]) => key.endsWith('Error'))
+  let errorClassesFromEntryPoint = Object.fromEntries(
+    Object.entries(importsFromEntryPoint).filter(
+      ([key, value]) => key.endsWith('Error') && value.toString().startsWith('class')
+    )
+  ) as any;
+  errorClassesFromEntryPoint = {
+    ...errorClassesFromEntryPoint,
+    MongoPoolClosedError,
+    MongoWaitQueueTimeoutError
+  };
+
+  const errorClassesFromErrorSrc = Object.fromEntries(
+    Object.entries(importsFromErrorSrc).filter(
+      ([key, value]) => key.endsWith('Error') && value.toString().startsWith('class')
+    )
   );
-  errorClasses = { ...errorClasses, MongoPoolClosedError, MongoWaitQueueTimeoutError };
 
-  for (const errorName in errorClasses) {
-    describe(errorName, () => {
-      it(`name should be read-only`, () => {
+  it('all defined errors should be public', () => {
+    expect(
+      setDifference(Object.keys(errorClassesFromEntryPoint), Object.keys(errorClassesFromErrorSrc))
+    ).to.have.property('size', 3);
+
+    expect(
+      setDifference(Object.keys(errorClassesFromErrorSrc), Object.keys(errorClassesFromEntryPoint))
+    ).to.have.property('size', 0);
+  });
+
+  describe('error names should be read-only', () => {
+    for (const [errorName, errorClass] of Object.entries(errorClassesFromEntryPoint)) {
+      it(`${errorName} should be read-only`, () => {
         // Dynamically create error class with message
-        let error = new errorClasses[errorName]('generated by test');
+        const error = new (errorClass as any)('generated by test', {});
         // expect name property to be class name
         expect(error).to.have.property('name', errorName);
 
         try {
           error.name = 'renamed by test';
+          // eslint-disable-next-line no-empty
         } catch (err) {}
         expect(error).to.have.property('name', errorName);
```

Comment:
Some drive by improvement we could consider, is being more direct about the assertions? I notice we have the eslint disable down here but the one at the top of the file is commented out. Maybe we remove the need for either.

```suggestion
        expect(errorClass.prototype).to.have.ownPropertyDescriptor('name')
          .that.has.property('set', undefined);
        expect(errorClass.prototype).to.have.ownPropertyDescriptor('name')
          .that.has.property('get').that.is.a('function');
```

Reply:
This assertion suggestion reads a bit weird to me, I feel like expecting a name property of a specified value is easier to understand than indirect references to getters and setters... the important thing, after all, is that the error has a name, not that it's specifically the outcome of a getter (imho)

Reply:
I feel like we had this exact convo when we first merged these tests 😅 , I'm fine with it as is too, if so then the ask would be to remove the eslint comment at the top of the file

Reply:
Done!